build.clj

```diff
@@ -78,6 +80,15 @@
           ns-decls (u/step "Determine compilation order for Metabase files"
                      (metabase-namespaces-in-topo-order basis))]
       (with-duration-ms [duration-ms]
+        ;; Clojure scans the source files to find tests. Compiling the uber jar
+        ;; changes the file structure, so we will scan for tests at compile time
+        ;; and store them in a file 'namespaces.txt' in the uberjar. At runtime,
+        ;; this file will be used to run the tests.
+        (b/write-file { :path (str class-dir "/namespaces.txt")
+                        :string (clojure.string/join \newline
+                                  (mapv #(subs (str %) 2)
+                                    (metabase.test-runner/find-tests ["local/src", "dev/src", "shared/test", "resources", "src", "test", "shared/src", "test_resources", "test_config"])))})
```

Comment:
Maybe here we can add an exclusion list like
```
(list 
   metabase.api.preview-embed-test 
   metabase.api.task-test)
```
or even better, individual functions
```
(remove nil? (list
              (if (= (tx/db-test-env-var-or-throw :ocient :mode "extended") "minimal")
                :metabase.api.preview-embed-test/query-max-results-constraint-test
                nil)
              :metabase.api.task-test/load-driver-namespace-race-condition-test))
```